src/routes/AuthLayoutRouter/AuthLayoutRouter.js

```diff
@@ -178,7 +182,12 @@ export default function AuthLayoutRouter (props) {
     return <Redirect to='/welcome' />
   }
 
-  if (currentGroupSlug && !currentGroup && !currentGroupLoading) {
+  if (!currentGroupMembership && hasDetail && paramPostId && currentGroupSlug) {
+    // this covers "Post I can see, and groupSlug for a group I cannot access." But it does not necessarily cover "Post I can see because its Public, but that isn't posted to any groups I have membership of"
+    return <Redirect push to={postUrl(paramPostId, { context: 'all', groupSlug: null })} />
+  }
+
+  if (currentGroupSlug && !currentGroup && !currentGroupLoading) { // this will have to be pushed down the hierarchy in the routers
```

Comment:
not sure what this comment is saying, didnt you just push it down already? is there more to do?

Reply:
No, figured that there isn't any merit in pushing it down the component tree but missed removing comment, soz